src/ledger.js

```diff
@@ -627,7 +634,94 @@ export class LedgerExportPublicKey extends LedgerExportHDNode {
       throw new Error("Received no public key from Ledger device.");
     }
   }
+}
+
+/**
+ * Class for wallet extended public key (xpub) interaction at a given BIP32 path.
+ * @extends {module:ledger.LedgerExportHDNode}
+ */
+export class LedgerExportExtendedPublicKey extends LedgerExportHDNode {
+  constructor({bip32Path, network}) {
+    super({bip32Path});
+    this.network = network;
+  }
+  
+  messages() {
+    const messages = super.messages();
 
+    if (this.hasBIP32PathWarning()) { 
+      messages.push({
+        image: IMAGES[LEDGER].exportPublicKeyBeta,
+        state: ACTIVE, 
+        level: INFO, 
+        version: ">=1.6.0",
+        text: `Your Ledger will ask to confirm "Export public key?"`,
+        code: "ledger.export.xpub",
+        action: LEDGER_RIGHT_BUTTON,
+      });
+
+      messages.push({
+        image: IMAGES[LEDGER].exportPublicKeyV1,
+        state: ACTIVE, 
+        level: INFO, 
+        version: "<1.6.0",
+        text: `Your Ledger will ask to confirm "Export public key?"`,
+        code: "ledger.export.xpub",
+        action: LEDGER_RIGHT_BUTTON,
+      });
+    }
+
+    return messages;
+  }
+
+  /**
+   * Get fingerprint for given path. This is useful for generating xpubs
+   * which need the fingerprint of the parent pubkey
+   * @param {string} path - path of pubkey to retrieve fingerprint for
+   * @returns {string} fingerprint
+   */
+   async getFingerprint(path) {
+    let fingerprint
+    await this.withApp(async (app) => {
+      const key = await app.getWalletPublicKey(path)
+      let compressed = compressPublicKey(key.publicKey)
+      compressed = Buffer.from(compressed, 'hex')
+      const hash = hash160(compressed)
+      fingerprint = ((hash[0] << 24) | (hash[1] << 16) | (hash[2] << 8) | hash[3]) >>> 0;
+    })
+    return fingerprint
+   }
+   
+  /**
+   * Retrieve extended public key(xpub) from Ledger device for a given BIP32 path
+   * @example
+   * import {LedgerExportExtendedPublicKey} from "unchained-wallets";
+   * const interaction = new LedgerExportExtendedPublicKey({network, bip32Path});
+   * const xpub = await interaction.run();
+   * console.log(xpub);
+   * @returns {string} extended public key(xpub) for the BIP32 path of a given instance
+   */
+  async run() {
+    const walletPublicKey = await super.run();
+    const compressedPubkey = compressPublicKey(walletPublicKey.publicKey)
+
+    let node;
+    await this.withApp(async (app) => {
+      // get the hd wallet node and fill in missing properties
+      node = bip32.fromPublicKey(Buffer.from(compressedPubkey, 'hex'),
```

Comment:
this probably could be pulled out into a `getHDWalletNode` utility function in `unchained-bitcoin` too.